pyas2/models.py

```diff
@@ -39,6 +43,36 @@ class PublicCertificate(models.Model):
         verbose_name=_('Verify Certificate'), default=True,
         help_text=_('Uncheck this option to disable certificate verification.'))
 
+    certificate_valid_from = models.DateTimeField(null=True, blank=True)
+    certificate_valid_to = models.DateTimeField(null=True, blank=True)
+
+    def extract_info(self):
+        '''
+        Extract validity information from the certificate and return a dictionnary
+        '''
+
+        der = pem_to_der(self.certificate)
+        cert = crypto.load_certificate(crypto.FILETYPE_ASN1, der[0])
+
+        cert_info = {'valid_from': None,
+                     'valid_to': None}
+
+        if der:
+            cert_info['valid_from'] = datetime.strptime(cert.get_notBefore().decode('utf8'), "%Y%m%d%H%M%SZ")
```

Comment:
I think you can also use the django.utils.dateparse module for doing this.

Reply:
The output of get_notBefore is: 
b'20190301181722Z' (or decoded: '20190301181722Z') 
for django.utils.dateparse to do correctly, the value would have to be: 
'2019-03-01 18:17:22Z'

Reply:
Okay